evm-ext/src/solidity_abi.rs

```diff
@@ -299,18 +369,50 @@ fn sol_to_value(sv: DynSolValue) -> Value {
         DynSolValue::FixedBytes(w, _size) => {
             Value::struct_(Struct::pack(vec![Value::vector_u8(w.to_vec())]))
         }
-        DynSolValue::Address(a) => Value::address(a.to_move_address()),
+        DynSolValue::Address(a) => match layout {
+            MoveTypeLayout::Address => Value::address(a.to_move_address()),
+            MoveTypeLayout::Signer => Value::master_signer(a.to_move_address()),
```

Comment:
Nice catch, address could be a signer.

Reply:
Actually, thinking about this again, it shouldn't be allowed to create a signer because then you could use this to forge a signature of any address. Since the main purpose of this API is to parse data coming back from EVM calls, there should never be a signer object to decode. I'll replace this with an error instead to make sure it cannot happen.

Reply:
Done in https://github.com/MovedNetwork/op-move/pull/354/commits/5f2f40a980db0f129317b5386fee15d411713df7